File: articles/institutional-norms-social-expectations/c/institutional_norms_score.c
#include <stdio.h>

double normative_stability_raw(
    double norm_repetition,
    double expectation_convergence,
    double internalization,
    double social_enforcement,
    double legitimacy_alignment,
    double trust_reinforcement,
    double role_clarity,
    double learning_capacity,
    double fragmentation_pressure,
    double unequal_normative_burden,
    double suppressive_pressure
) {
    return
        0.13 * norm_repetition +
        0.14 * expectation_convergence +
        0.13 * internalization +
        0.11 * social_enforcement +
        0.13 * legitimacy_alignment +
        0.11 * trust_reinforcement +
        0.09 * role_clarity +
        0.08 * learning_capacity -
        0.13 * fragmentation_pressure -
        0.10 * unequal_normative_burden -
        0.08 * suppressive_pressure;
}

int main(void) {
    double score = normative_stability_raw(
        82.0, 84.0, 78.0, 76.0, 80.0, 79.0, 77.0, 74.0,
        22.0, 20.0, 18.0
    );

    printf("Normative stability raw score: %.2f\n", score);
    return 0;
}

File: articles/institutional-norms-social-expectations/fortran/institutional_norms_score.f90
program institutional_norms_score
  implicit none

  real :: norm_repetition, expectation_convergence, internalization
  real :: social_enforcement, legitimacy_alignment, trust_reinforcement
  real :: role_clarity, learning_capacity, fragmentation_pressure
  real :: unequal_normative_burden, suppressive_pressure
  real :: score

  norm_repetition = 82.0
  expectation_convergence = 84.0
  internalization = 78.0
  social_enforcement = 76.0
  legitimacy_alignment = 80.0
  trust_reinforcement = 79.0
  role_clarity = 77.0
  learning_capacity = 74.0
  fragmentation_pressure = 22.0
  unequal_normative_burden = 20.0
  suppressive_pressure = 18.0

  score = 0.13 * norm_repetition + &
          0.14 * expectation_convergence + &
          0.13 * internalization + &
          0.11 * social_enforcement + &
          0.13 * legitimacy_alignment + &
          0.11 * trust_reinforcement + &
          0.09 * role_clarity + &
          0.08 * learning_capacity - &
          0.13 * fragmentation_pressure - &
          0.10 * unequal_normative_burden - &
          0.08 * suppressive_pressure

  print *, "Normative stability raw score:", score
end program institutional_norms_score
